### PR TITLE
Updated vulnerable Nuget packages

### PR DESCRIPTION
**Pull request by LACrossley**

### CODE REVIEW

**Review by karen-hanna — DISMISSED** [review]

**Review by furqanagwan — DISMISSED** [review]

File: azure/template.json [hunk 1]
```diff
@@ -52,11 +52,12 @@
...
-    "variables": {
-        "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
-        "resourceNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName'),'-', parameters('serviceName')))]",
-        "appServiceName": "[concat(variables('resourceNamePrefix'), '-as')]"
-    },
+  "variables": {
+    "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
+    "resourceNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName'),'-', parameters('serviceName')))]",
+    "appServiceName": "[concat(variables('resourceNamePrefix'), '-as')]",
+    "adminServiceConfigNames": "SFA.DAS.AdminService"
```
Comment (ChrisJWoodcock) [inline azure/template.json:59]: This is odd - see my comment below about the DfeSignIn config row key - which I'm sure the system is currently using - but if it was not here already how - was it using it?

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.AdminService.Web/SFA.DAS.AdminService.Web.csproj [hunk 2]
```diff
@@ -40,31 +41,37 @@
...
+    <PackageReference Include="Microsoft.IdentityModel.JsonWebTokens" Version="7.6.0" />
+    <PackageReference Include="Microsoft.IdentityModel.KeyVaultExtensions" Version="7.6.0" />
+    <PackageReference Include="Microsoft.IdentityModel.Logging" Version="7.6.0" />
+    <PackageReference Include="Microsoft.IdentityModel.Protocols" Version="7.6.0" />
+    <PackageReference Include="Microsoft.IdentityModel.Protocols.OpenIdConnect" Version="7.6.0" />
+    <PackageReference Include="Microsoft.IdentityModel.Protocols.WsFederation" Version="7.6.0" />
     <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="6.0.8" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
 
 		<PackageReference Include="NLog.Web.AspNetCore" Version="5.1.4" />
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.AdminService.Web/SFA.DAS.AdminService.Web.csproj:56]: This is an example of where I would update the white - as this is horrible formatting

File: src/SFA.DAS.AdminService.Web/Startup.cs [hunk 3]
```diff
@@ -50,14 +50,13 @@ namespace SFA.DAS.AdminService.Web
...
-        private const string Version = "1.0";
+
         public IConfiguration Configuration { get; }
         public IWebConfiguration ApplicationConfiguration { get; set; }
 
-        public Startup(IConfiguration configuration, IHostingEnvironment env, ILogger<Startup> logger)
+        public Startup(IConfiguration configuration, IHostEnvironment env, ILogger<Startup> logger)
         {
             _env = env;
             _logger = logger;
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.AdminService.Web/Startup.cs:62]: #if DEBUG
            if (!configuration.IsDev())
            {
                config.AddJsonFile("appsettings.json", false)
                    .AddJsonFile("appsettings.Development.json", true);
            }
#endif

This section is causing some confusion - I think it does not need to explicitly state the appSettings.json should be read as it is always read by default - it being read by default even though this read is DEBUG only is why the SFA.DAS.Provider.DfeSignIn is actually being read currently and the application works.

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.AdminService.Web/SFA.DAS.AdminService.Web.csproj [hunk 4]
```diff
@@ -15,65 +15,74 @@
...
+			<PrivateAssets>all</PrivateAssets>
+			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+		</PackageReference>
+		<PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="6.0.8">
+			<PrivateAssets>all</PrivateAssets>
+			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+		</PackageReference>
+		<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="6.0.0" />
+		<PackageReference Include="Microsoft.IdentityModel.Logging" Version="7.6.0" />
+		<PackageReference Include="NuGet.CommandLine" Version="6.9.1">
```
Comment (karen-hanna) [inline src/SFA.DAS.AdminService.Web/SFA.DAS.AdminService.Web.csproj:36]: image

**Review by karen-hanna — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.AdminService.Application.Interfaces/SFA.DAS.AdminService.Application.Interfaces.csproj [hunk 5]
```diff
@@ -6,8 +6,14 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Azure.Identity" Version="1.12.0-beta.3" />
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AdminService.Application.Interfaces/SFA.DAS.AdminService.Application.Interfaces.csproj]: confirmed that this has been updated

File: src/SFA.DAS.AdminService.Web/appsettings.json [hunk 6]
```diff
@@ -8,7 +8,7 @@
     }
   },
   "ConfigurationStorageConnectionString": "UseDevelopmentStorage=true;",
-  "ConfigNames": "SFA.DAS.Provider.DfeSignIn",
+  "ConfigNames": "SFA.DAS.AdminService",
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AdminService.Web/appsettings.json]: confirmed both are now included

File: src/SFA.DAS.AdminService.Web/Startup.cs [hunk 7]
```diff
@@ -114,6 +108,10 @@ public void ConfigureServices(IServiceCollection services)
                 options.RequestCultureProviders.Clear();
             });
 
+            services.AddFluentValidationAutoValidation();
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AdminService.Web/Startup.cs]: has been updated 
`services
    .AddFluentValidationAutoValidation()
    .AddFluentValidationClientsideAdapters()
    .AddValidatorsFromAssemblyContaining<Startup>();`

File: src/SFA.DAS.AdminService.Web.UnitTests/SFA.DAS.AdminService.Web.UnitTests.csproj [hunk 8]
```diff
@@ -5,7 +5,10 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="AutoFixture" Version="4.18.0" />
+    <PackageReference Include="Azure.Core" Version="1.40.0" />
+    <PackageReference Include="Azure.Identity" Version="1.12.0-beta.3" />
     <PackageReference Include="FluentAssertions" Version="6.10.0" />
+    <PackageReference Include="Microsoft.Data.SqlClient" Version="5.2.0" />
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AdminService.Web.UnitTests/SFA.DAS.AdminService.Web.UnitTests.csproj]: transient references needed updated for warnings

File: src/SFA.DAS.AdminService.Application.Interfaces/SFA.DAS.AdminService.Application.Interfaces.csproj [hunk 9]
```diff
@@ -6,8 +6,14 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Azure.Identity" Version="1.12.0" />
```
[the remark this answers is not in the document]
Comment (LACrossley) [reply src/SFA.DAS.AdminService.Application.Interfaces/SFA.DAS.AdminService.Application.Interfaces.csproj:14]: Done

File: azure/template.json [hunk 10]
```diff
@@ -117,45 +118,53 @@
...
+                                },
+                                {
+                                  "name": "WEBSITE_SWAP_WARMUP_PING_STATUSES",
+                                  "value": "200"
+                                },
+                                {
+                                  "name": "ConfigNames",
+                                  "value": "[variables('adminServiceConfigNames')]"
+                                },
+                                {
```
[the remark this answers is not in the document]
Comment (LACrossley) [reply azure/template.json:158]: Done, I think

File: src/SFA.DAS.AdminService.Application.Interfaces/SFA.DAS.AdminService.Application.Interfaces.csproj [hunk 11]
```diff
@@ -6,8 +6,18 @@
...
+  <ItemGroup>
+	<!--These packages are required to override transient package vulnerabilities-->
+    <PackageReference Include="Azure.Identity" Version="1.12.0" />
+    <PackageReference Include="EPPlus" Version="6.0.7" />
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.61.3" />
+    <PackageReference Include="Microsoft.IdentityModel.Abstractions" Version="7.6.0" />
+    <PackageReference Include="System.Drawing.Common" Version="8.0.4" />
+    <PackageReference Include="System.Security.Cryptography.Xml" Version="8.0.0" />
+    <PackageReference Include="System.Text.RegularExpressions" Version="4.3.1" />
+  </ItemGroup>
```
[the remark this answers is not in the document]
Comment (LACrossley) [reply src/SFA.DAS.AdminService.Application.Interfaces/SFA.DAS.AdminService.Application.Interfaces.csproj:21]: Updated the package.

File: src/SFA.DAS.AdminService.Common/SFA.DAS.AdminService.Common.csproj [hunk 12]
```diff
@@ -24,6 +24,8 @@
     <PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
     <PackageReference Include="Microsoft.AspNetCore.Authorization" Version="6.0.8" />
+    <PackageReference Include="System.Net.Http" Version="4.3.4" />
+    <PackageReference Include="System.Text.RegularExpressions" Version="4.3.1" />
```
[the remark this answers is not in the document]
Comment (LACrossley) [reply src/SFA.DAS.AdminService.Common/SFA.DAS.AdminService.Common.csproj:28]: Ah yes! Amended.

File: src/SFA.DAS.AdminService.Infrastructure/SFA.DAS.AdminService.Infrastructure.csproj [hunk 13]
```diff
@@ -1,13 +1,22 @@
...
-    <PackageReference Include="MediatR.Extensions.Microsoft.DependencyInjection" Version="11.1.0" />
-    <PackageReference Include="Microsoft.AspNetCore.Http.Features" Version="5.0.17" />
+    <!--These packages are required to override transient package vulnerabilities-->
+	<FrameworkReference Include="Microsoft.AspNetCore.App" />
+	<PackageReference Include="Azure.Identity" Version="1.12.0" />
+	<PackageReference Include="Microsoft.Identity.Client" Version="4.61.3" />
+	<PackageReference Include="Microsoft.IdentityModel.Abstractions" Version="7.6.0" />
+	<PackageReference Include="System.Drawing.Common" Version="8.0.4" />
+	<PackageReference Include="System.Security.Cryptography.Xml" Version="8.0.0" />
+	<PackageReference Include="System.Text.RegularExpressions" Version="4.3.1" />
```
[the remark this answers is not in the document]
Comment (LACrossley) [reply src/SFA.DAS.AdminService.Infrastructure/SFA.DAS.AdminService.Infrastructure.csproj:15]: Updated.

File: src/SFA.DAS.AdminService.Settings/SFA.DAS.AdminService.Settings.csproj [hunk 14]
```diff
@@ -1,15 +1,25 @@
...
 
+  <ItemGroup>
+	<!--These packages are required to override transient package vulnerabilities-->
+	<PackageReference Include="Azure.Identity" Version="1.12.0" />
+	<PackageReference Include="Azure.Storage.Common" Version="12.18.1" />
+	<PackageReference Include="Microsoft.Identity.Client" Version="4.61.3" />
+	<PackageReference Include="Microsoft.IdentityModel.Abstractions" Version="7.6.0" />
+	<PackageReference Include="System.Drawing.Common" Version="8.0.4" />
+	<PackageReference Include="System.Security.Cryptography.Xml" Version="8.0.0" />
+	<PackageReference Include="System.Text.RegularExpressions" Version="4.3.1" />
```
[the remark this answers is not in the document]
Comment (LACrossley) [reply src/SFA.DAS.AdminService.Settings/SFA.DAS.AdminService.Settings.csproj:16]: Updated.

File: src/SFA.DAS.AdminService.Web.UnitTests/SFA.DAS.AdminService.Web.UnitTests.csproj [hunk 15]
```diff
@@ -3,6 +3,11 @@
     <ProjectGuid>F07B3326-5CF7-4CFC-90FE-38606A7425B8</ProjectGuid>
     <TargetFramework>net6.0</TargetFramework>
   </PropertyGroup>
+  <ItemGroup>
+	<!--These packages are required to override transient package vulnerabilities-->
+    <PackageReference Include="Azure.Core" Version="1.40.0" />
+    <PackageReference Include="Azure.Identity" Version="1.12.0" />
+  </ItemGroup>
```
[the remark this answers is not in the document]
Comment (LACrossley) [reply src/SFA.DAS.AdminService.Web.UnitTests/SFA.DAS.AdminService.Web.UnitTests.csproj:10]: Updated.

File: src/SFA.DAS.AdminService.Web/SFA.DAS.AdminService.Web.csproj (hunk 4 above)
Comment (LACrossley) [reply src/SFA.DAS.AdminService.Web/SFA.DAS.AdminService.Web.csproj:36]: Updated.

File: src/SFA.DAS.AdminService.Web/Startup.cs (hunk 3 above)
Comment (LACrossley) [reply src/SFA.DAS.AdminService.Web/Startup.cs:62]: Ok. Amended.

File: src/SFA.DAS.AdminService.Web/SFA.DAS.AdminService.Web.csproj [hunk 16]
```diff
@@ -16,21 +16,22 @@
...
-    <PackageReference Include="AutoMapper" Version="11.0.1" />
+    <PackageReference Include="MediatR" Version="12.2.0" />
+    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.22.0" />
+    <PackageReference Include="AutoMapper" Version="13.0.1" />
     <PackageReference Include="Microsoft.AspNetCore.DataProtection.StackExchangeRedis" Version="6.0.8" />
-    <PackageReference Include="Microsoft.ApplicationInsights.NLogTarget" Version="2.21.0" />
+    <PackageReference Include="Microsoft.ApplicationInsights.NLogTarget" Version="2.22.0" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.WsFederation" Version="6.0.8" />
-    <PackageReference Include="Microsoft.Azure.Services.AppAuthentication" Version="1.6.2" />
+    <PackageReference Include="Microsoft.Data.SqlClient" Version="5.2.0" />
```
[the remark this answers is not in the document]
Comment (LACrossley) [reply src/SFA.DAS.AdminService.Web/SFA.DAS.AdminService.Web.csproj]: Updated to resolve a transient package vulnerability.

File: azure/template.json (hunk 1 above)
Comment (LACrossley) [reply azure/template.json]: Resolved, and included.

[the remark this answers is not in the document]
Comment (LACrossley) [reply azure/template.json]: Done.

File: azure/template.json [hunk 17]
```diff
@@ -117,45 +118,53 @@
...
-                              },
-                              {
-                                "name": "WEBSITE_SWAP_WARMUP_PING_STATUSES",
-                                "value": "200"
-                              }
-                          ]
-                        }
+                            "array": [
+                                {
+                                  "name": "ConfigurationStorageConnectionString",
```
[the remark this answers is not in the document]
Comment (LACrossley) [reply azure/template.json]: Corrected.

File: src/SFA.DAS.AdminService.Web/SFA.DAS.AdminService.Web.csproj (hunk 2 above)
Comment (LACrossley) [reply src/SFA.DAS.AdminService.Web/SFA.DAS.AdminService.Web.csproj]: Updated, I think.

**Comment by ChrisJWoodcock** [comment]
@karen-hanna use this as a reference only PR

**Pull request closed without merging**